trigger-actions-framework/main/default/classes/TriggerActionFlowBypassProcessor.cls

```diff
@@ -0,0 +1,48 @@
+/*
+   Copyright 2022 Google LLC
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+public inherited sharing abstract class TriggerActionFlowBypassProcessor {
+	public static final String INVALID_TYPE = 'The bypassType must equal Object, Apex, or Flow';
+	public static final String APEX_STRING = 'Apex';
+	public static final String FLOW_STRING = 'Flow';
+	public static final String OBJECT_STRING = 'Object';
+	private static final Set<String> REQUEST_TYPES = new Set<String>{
+		OBJECT_STRING,
+		APEX_STRING,
+		FLOW_STRING
+	};
+
+	protected Boolean hasActionBeenPerformed = true;
+
+	public Boolean hasActionBeenPerformed(
+		String requestType,
+		String requestName
+	) {
+		if (!REQUEST_TYPES.contains(requestType)) {
+			throw new IllegalArgumentException(INVALID_TYPE);
+		} else if (requestType == FLOW_STRING) {
+			this.processFlowBypasses(requestName);
+		} else if (requestType == APEX_STRING) {
+			this.processApexBypasses(requestName);
+		} else if (requestType == OBJECT_STRING) {
+			this.processObjectBypasses(requestName);
+		}
+		return this.hasActionBeenPerformed;
```

Comment:
It seems like the only time we actually need to return anything is with the `isBypassed` action. Maybe I am not understanding this correctly, but this variable name is confusing me. Curious to hear your thoughts.

Reply:
`hasActionBeenPerformed` is always true for the other classes. Which variable name are you referring to?

Reply:
This public method signature is the basis for the following invocable actions:
- Bypass
- ClearBypass
- ClearAllBypasses
- IsBypassed

Each of these are of `void` return type with the exception of `IsBypassed`. 

Within the implementation of `IsBypassed`, the value of `hasActionBeenPerformed` is set to true and returned. This is a little confusing when the input class, flow, or sObject is _not_ bypassed, the value of `hasActionBeenPerformed` will be false, although the action has indeed been performed. In all other cases, we are returning a boolean which is immediately orphaned at runtime - slightly odd behavior. The returning of a value (which can be immediately orphaned) but calling it `hasActionBeenPerformed` makes me wonder if what we are trying to achieve requires a structure like:

```java
public class InvocableBypassResult {
  public Boolean haBeenPerformed;
  public Boolean value;
}
```
Seems a little overkill, but the questioning feels a natural consequence of the current naming conventions. I think a simple Boolean response is sufficient, but perhaps with a name change is necessary. Classic "what to name this" concerns 🙃 
 
When I think of language for this type of setup, the first verb and noun which come to mind are "execute" for the method name and "result" for the type of the response.
```java
Boolean response = new TAFBypassProcessor().execute(bypassType);
```
 It's not exactly original or flashy - I am open to suggestions here.

Reply:
The public method signature need not be tied in any strong way to the method signatures for any class or dependent class's internals. `hasActionBeenPerformed` also defaults to true. Perhaps `hasBeenBypassed` would make the intent clearer?

As for the "orphaning" of the Boolean, while this is a common pattern when only some subclasses care to define an actual return type, I've pushed an alternate up to this branch - a bit less object-oriented at the call site in `TriggerActionFlowIsBypassed`, but perhaps a better separation of concerns.

On the topic of naming - I've never seen instance methods prefaced with `this` in their method names before, such as in `TriggerActionFlow`'s `thisFlowIsBypassed`. Is there a reason for this unusual convention? It's very atypical compared to the standard prefacing of instance methods with `this.` (eg `this.flowIsBypassed()` when calling that method)

Reply:
Great call on the `thisFlowIsBypassed` - definitely an opportunity to clean that one up. Searching through the repository shows that `thisFlowIsBypassed` is the only place with the `this` prefix.  Not sure what I was thinking there... I can raise a separate PR to update and keep the scope of this change well defined.